src/constraints.jl

```diff
@@ -1542,3 +1542,22 @@ function relax_with_penalty!(
 ) where {T}
     return relax_with_penalty!(model, Dict(); default = default)
 end
+
+function parse_constraint_head(error_fn::Function, ::Val{:(:=)}, lhs, rhs)
+    new_lhs, parse_code_lhs = _rewrite_expression(lhs)
+    new_rhs, parse_code_rhs = _rewrite_expression(rhs)
+    parse_code = quote
+        $parse_code_lhs
+        $parse_code_rhs
+    end
+    build_code = quote
+        if $new_rhs isa Bool
```

Comment:
Since we do it at the parsing level, it won't work if `rhs` is a Julia variable (not a JuMP decision variable) for which the value is a `Bool` right ?

Reply:
This part is at runtime. See the tests.

Reply:
Ah indeed. I guess it's best to add code inside the `build_constraint` that add code that is generated by the macro if possible

Reply:
But in this case, it might be difficule

Reply:
We would need a new JuMP set `Equal` with and do `build_constraint(error, (a, b), JuMP.Equal()`